From 19fa3fa5e8ef045b31434d17799fdffceb8a49f8 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:14:37 -0500
Subject: [ethernet] r8169: add disable/enable RTL8168G pll function

Message-id: <1416413687-12219-7-git-send-email-ivecera@redhat.com>
Patchwork-id: 100358
O-Subject: [RHEL7.1 PATCH 06/16] r8169:add disable/enable RTL8168G pll function
Bugzilla: 1165764
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

BZ: #1165764

Upstream commit(s):
commit b8e5e6ad7115befef13a4493f1d2b8e438abc058
Author: Chun-Hao Lin <hau@realtek.com>
Date:   Wed Oct 1 23:17:13 2014 +0800

    r8169:add disable/enable RTL8168G pll function

    RTL8168G also can disable/enable pll function.

    Signed-off-by: Chun-Hao Lin <hau@realtek.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 44cdc86..5e84af9 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -4375,6 +4375,7 @@ static void r8168_pll_power_down(struct rtl8169_private *tp)
  case RTL_GIGA_MAC_VER_41:
   rtl_w1w0_eri(tp, 0x1a8, ERIAR_MASK_1111, 0x00000000,
         0xfc000000, ERIAR_EXGMAC);
+  RTL_W8(PMCH, RTL_R8(PMCH) & ~0x80);
   break;
  }
 }
@@ -4399,6 +4400,7 @@ static void r8168_pll_power_up(struct rtl8169_private *tp)
   break;
  case RTL_GIGA_MAC_VER_40:
  case RTL_GIGA_MAC_VER_41:
+  RTL_W8(PMCH, RTL_R8(PMCH) | 0xc0);
   rtl_w1w0_eri(tp, 0x1a8, ERIAR_MASK_1111, 0xfc000000,
         0x00000000, ERIAR_EXGMAC);
   break;
-- 
1.7.1